src/app/components/play-round/play-round.component.html

```diff
@@ -10,7 +21,8 @@
   mat-raised-button
   class="interact-button"
   color="primary"
-  *ngIf="this.foundRoundMarker$ | async as marker" [disabled]="!marker.active"
+  *ngIf="this.foundRoundMarker$ | async as marker"
+  [disabled]="!marker.active || ((this.studentRoundService.currentBeePollen$ | async) >= 3 && !marker.isNest)"
```

Comment:
Seems like everything works out just fine with the security rules!

There is one weird thing, though. At the moment, the "interact" button depends on the `currentPollenCount$` observable to know whether it should be enabled, but that observable takes a brief moment to resolve, and until it gives back a value, the "interact" button is enabled; If you click it during that window of time, you can acquire more than 3 pollens at a time.

What do you think about something like this &hellip;?

```html
<!--
  Make sure the currentBeePollen$ observable is resolved before we try to
  display the button (so we know whether it should be enabled or not).
-->
<ng-container *ngIf="{ count: (this.studentRoundService.currentBeePollen$ | async) } as currentBeePollen">
  <button
    mat-raised-button
    class="interact-button"
    color="primary"
    *ngIf="this.foundRoundMarker$ | async as marker"
    [disabled]="!marker.active || currentBeePollen.count >= 3 && !marker.isNest"
    (click)="clickInteract(marker)"
  >
    Visit {{marker.name}}
  </button>
</ng-container>
```

Here, we don't display the interact button until the `currentPollenCount$` observable resolves. It's a bit hacky, though. If you have a better way to do this—or, alternatively, if you think it's not worth the hassle—that would be great!

Reply:
For example, if you have a script that clicks a few hundred times really fast, you can do something like this: :upside_down_face:

![The student side, where the student has two much pollen, and the pollen counter is filling up half of the screen](https://user-images.githubusercontent.com/56209343/93530822-d57a4680-f903-11ea-897d-196495dc855f.png)

Reply:
By the way, the way that you can reproduce the "button enabled for just a moment" effect is to go to a blooming flower, get 3 pollens, and then quickly move the blooming flower on- and off-camera. When the blooming flower enters the camera's field of view, the button will be enabled for just a second.

Reply:
Part of the thing here is that in the future we will also want to not allow students to simply interact with the same flower 3 times in fast succession so we will need some kind of debounce based on the last flower interacted with or on the interact button itself.

Reply:
Oh, sure, I hear you! That's a bit of a separate issue, though. Even with debouncing, you could still collect 4, 5… pollen by moving around to different flowers and clicking during the brief window when the button is enabled. (My hundreds-of-pollen script would no longer be viable, but that was more of a proof of concept anyway 🙃)